src/redux/books/booksSlice.js

```diff
@@ -29,9 +48,42 @@ export const booksSlice = createSlice({
       state.books.push(action.payload);
     },
     remove: (state, action) => {
-      state.books = state.books.filter((book) => book.id !== action.payload);
+      state.books = state.books.filter((book) => book.item_id !== action.payload);
     },
   },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        state.status = 'succeded';
+        const liveBooks = action.payload;
+        const booksStore = [];
+        Object.keys(liveBooks).map((id) => (
+          booksStore.push(
+            {
+              item_id: id,
+              title: liveBooks[id][0].title[0],
+              author: liveBooks[id][0].author[0],
+              category: liveBooks[id][0].category,
+            },
+          )
+        ));
+        state.books = booksStore;
+      })
+      .addCase(fetchBooks.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      })
+      .addCase(addBooks.fulfilled, async (state) => {
+        state.status = 'bookadded';
+      })
+      .addCase(addBooks.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      })
```

Comment:
## [OPTIONAL]
- [ ] I think your redux may look good if you work on your error handling, currently when a book is created the API response is correct but an error is dispatched in your redux store.